pkg/ipam/node_manager.go

```diff
@@ -279,6 +285,12 @@ func (n *NodeManager) Update(resource *v2.CiliumNode) (nodeSynced bool) {
 			logLimiter:          logging.NewLimiter(10*time.Second, 3), // 1 log / 10 secs, burst of 3
 		}
 
+		if !n.instancesAPI.HasInstance(resource.InstanceID()) {
+			if _, ok := n.instancesAPIResync(context.TODO()); !ok {
+				node.logger().Warning("Failed to resync instance API after new node was found")
```

Comment:
> Thanks for the fix! Could you update the commit msg to be more descriptive on what the fix is doing and why?

The updated commit message
```
eni: fix new node not immediately triggers the creation of ENI

1. Sync the instances network interface information from instances api when a new node is created
but not in `instanceManager.instances`. This is fix errors in `ResyncInterfacesAndIPs` "Instance not found! Please delete
corresponding ciliumnode if instance has already been deleted" when new node
created in cloud provider then add to cluster and before resync triggered.

2. Delete instance from `instanceManager.instances` on node deleted. This will cause `NodeManager.Update()` to
invoke `instancesAPIResync` if this instance rejoins the cluster.
This ensures that `Node.recalculate()` does not use stale data for
instances which rejoin the cluster after their EC2 configuration has changed.

Fixes: #20678
```